File: 2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        rest = 0
        if(l1.val + l2.val > 9):
            start = ListNode(l1.val + l2.val - 10)
            rest = 1;
        else:
            start = ListNode(l1.val + l2.val)
        tmp = start
        while((l1.next != None) | (l2.next != None)):
            l1 = l1.next if l1.next else ListNode(0)
            l2 = l2.next if l2.next else ListNode(0)
            if (l1.val + l2.val + rest > 9):
                newtmp = ListNode(l1.val + l2.val + rest - 10)
                rest = 1
            else:
                newtmp = ListNode(l1.val + l2.val + rest)
                rest = 0
            tmp.next = newtmp
            tmp = newtmp
        if(rest):
            newtmp = ListNode(rest)
            tmp.next = newtmp
        return start

File: 2.js
/*
 * @lc app=leetcode id=2 lang=javascript
 *
 * [2] Add Two Numbers
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var HEAD = new ListNode();
    var pre = HEAD;
    var temp = 0;
    while((l1 != null) || (l2 != null) || (temp > 0)){
      var l1Val = l1 && l1.val || 0;
      var l2Val = l2 && l2.val || 0;
      var addResult = l1Val + l2Val + temp;
      temp = 0;
      if(addResult > 9){
        temp = 1;
        addResult = addResult - 10
      }
      var newNode = new ListNode(addResult);
      pre.next = newNode;
      pre = newNode;
      l1 && (l1 = l1.next);
      l2 && (l2 = l2.next);
    };
    return HEAD.next;
};
// @lc code=end

